onnx/external_data_helper.py

```diff
@@ -256,3 +258,42 @@ def write_external_data_tensors(model, filepath):  # type: (ModelProto, Text) ->
             tensor.ClearField(str('raw_data'))
 
     return model
+
+
+def save_model_to_external_data(model, model_file_path, all_tensors_to_one_file=True, location=None, size_threshold=1024,
+                                convert_attribute=False):  # type: (ModelProto, Union[IO[bytes], Text], bool, Optional[Text], int, bool) -> None
+    """
+    Call to set tensors with raw data as external data and save to a specified path.
+
+    @params
+    model: ModelProto to be converted.
+    model_file_path: A string containing a file name
+    all_tensors_to_one_file: If true, save all tensors to one external file specified by location.
+                             If false, save each tensor to a file named with the tensor name.
+    location: specify the external file that all tensors to save to.
+              If not specified, will use the model name.
+    size_threshold: Threshold for size of data. Only when tensor's data is >= the size_threshold
+    it will be converted to external data. To convert every tensor with raw data to external data set size_threshold=0.
+    convert_attribute: If true, convert all tensors to external data
+                       If false, convert only non-attribute tensors to external data
+    """
+    if convert_attribute:
+        convert_model_to_external_data(model, all_tensors_to_one_file, location, size_threshold)
+    else:
+        if all_tensors_to_one_file:
```

Comment:
why not update the convert_model_to_external_data to take convert_attribute... this way you dont need to duplicate code

Reply:
I like this idea, does not require the new function save_model_to_external_data. Could introduce a local variable that takes value _get_all_tensors(model) or _get_initializer_tensors(model) based on convert_attribute.
My sense was the original convert_model_to_external_data did the honors for all tensors .. should that not translate to a default value of True for convert_attribute?

Reply:
The idea behind introducing external data was to enable models whose size exceeds the 2G max limit set by protobuf also using external data for tensors can help with sharing weights between models... however serializing attributes to external data defeats the purpose, usually attributes consume little memory and serializing them will require more memory as now you will be storing all the external data info in the tensor proto as opposed to the actual data which will most likely be an int/float/bool
This also causes issues in runtimes as attributes are required for shape inferencing etc... serializing attributes by default is a bug and we should fix that and make it optional. Ideally the option to serialize attributes is not required for the reasons stated above but we planned to add it to support any existing user scenarios which we may not be aware of.

Reply:
Modified to remove the duplicate code and update the existing code to take the convert_attribute and file path.